packages/connection/legacy/class-jetpack-ixr-client.php

```diff
@@ -37,9 +37,11 @@ public function __construct( $args = array(), $path = false, $port = 80, $timeou
 		$defaults = array(
 			'url'     => $connection->xmlrpc_api_url(),
 			'user_id' => 0,
+			'headers' => array(),
 		);
 
-		$args = wp_parse_args( $args, $defaults );
+		$args            = wp_parse_args( $args, $defaults );
+		$args['headers'] = array_merge( array( 'Content-Type' => 'text/xml' ), (array) $args['headers'] );
```

Comment:
I note this has the parameters in the opposite order compared to r215538-wpcom, which means this allows the caller to override the content type while r215538-wpcom doesn't. Personally I like the added flexibility here, but I wanted to note the discrepancy.

Reply:
I didn't think of that until putting together the JP patch. I'll flip them around on wp.com if this is approved.

Reply:
D51896-code

Reply:
r216016-wpcom